src/preprocessing/passes/miplib_trick.cpp

```diff
@@ -550,7 +550,6 @@ PreprocessingPassResult MipLibTrick::applyInternal(
                   << "unexpected solution from arith's ppAssert()";
               Assert(nullMap.empty())
                   << "unexpected substitution from arith's ppAssert()";
-              d_preprocContext->addModelSubstitution(*ii, newVar.eqNode(one));
```

Comment:
My justification for removing it is that we now have the invariant `top-level substitutions = model substitutions`.  Either:
(A) This preprocessing pass was already maintaining this invariant, in which case this call is spurious and can be removed, since after this PR, this is a consequence of its call to adding top-level substitutions (https://github.com/cvc5/cvc5/blob/master/src/preprocessing/passes/miplib_trick.cpp#L592).
(B) This preprocessing pass is not maintaining this invariant, in which case we shouldn't have been making this call in the first place, since this would imply that we weren't keeping our model consistent with our preprocessing.

I don't quite follow what this pass is doing, but regardless, in either case this line should be deleted.